agent.py

```diff
@@ -342,6 +343,15 @@ def run(self, config=None):
                 try:
                     self.sd_backend.reload_check_configs = get_config_store(
                         self._agentConfig).crawl_config_template()
+
+                    # JMXFetch restarts should prompt reload
+                    try:
+                        jmx_launch = JMXFetch._get_jmx_launchtime()
+                        if self.last_jmx_piped and self.last_jmx_piped < jmx_launch:
+                            self.sd_backend.reload_check_configs = True
```

Comment:
I'm not sure this jmxfetch-specific code block should be here exactly: `JMXFetch` restarts aren't really related to the updates in the config template store (which is what the parent block is about).

On top of that, when JMXFetch restarts, it'd be nice to reload only the jmx-related check configs instead of all the check configs.

So:
1. Could we move this logic to outside this `if` block (i.e. right after `self.reload_configs_flag = False`)?
2. To avoid reloading all the check configs, we could do what's done here: https://github.com/DataDog/dd-agent/blob/5.15.0/agent.py#L281-L285 (which could be out in a separate function) instead of setting `self.sd_backend.reload_check_configs = True`

Let me know how that sounds to you, I might have missed something.